### DISCUSSION TITLE
getStaticProps called at runtime for the first time?

### DISCUSSION THREAD

**Discussion by balazsorban44** [vercel/next.js#11134] [Help] [answered]
According to the docs:
https://nextjs.org/docs/basic-features/data-fetching#when-should-i-use-getstaticprops

> The data required to render the page is available at build time ahead of a user’s request.

It sounds like that `getStaticProps` should fetch the data at build time.

In the beginning, there is no corresponding data (json) in the `build` folder for a page. On the first request (while running with `next start`), I go and fetch it from my data source first, populate a file then only after that, the subsequent requests will use this file.

I have more details on the issue, but first I would like to clarify if this is intentional behavior, or I do something wrong? In the latter case, I can try creating a reproduction example.

**Comment by timneutkens** [comment, marked as answer]
Depends on if you use fallback behavior: https://nextjs.org/docs/basic-features/data-fetching#fallback-true

**Reply by balazsorban44** [reply]
Of course, thank you... Feel dumb slipping over this, as I have followed the RFC from early on. 😅
